src/Domain/JacobianDiagnostic.cpp

```diff
@@ -0,0 +1,79 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/JacobianDiagnostic.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"     // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace domain {
+template <size_t Dim, typename Fr>
+void jacobian_diagnostic(
+    const gsl::not_null<Scalar<DataVector>*> jacobian_diagnostic,
+    const Jacobian<DataVector, Dim, typename Frame::ElementLogical, Fr>&
+        analytic_jacobian,
+    const TensorMetafunctions::prepend_spatial_index<
+        tnsr::I<DataVector, Dim, Fr>, Dim, UpLo::Lo,
+        typename Frame::ElementLogical>& numeric_jacobian_transpose) {
+  auto jacobian_difference_sum =
+      make_with_value<tnsr::i<DataVector, Dim, typename Frame::ElementLogical>>(
+          get<0, 0>(analytic_jacobian), 0.0);
+  auto numerical_difference_sum =
+      make_with_value<tnsr::i<DataVector, Dim, typename Frame::ElementLogical>>(
+          get<0, 0>(analytic_jacobian), 0.0);
+  constexpr double eps_to_avoid_div_by_zero = 1.e-20;
+
+  // i_hat = logical frame index
+  // i = mapped frame index
+  for (size_t i_hat = 0; i_hat < Dim; ++i_hat) {
+    for (size_t i = 0; i < Dim; ++i) {
+      jacobian_difference_sum.get(i_hat) -= analytic_jacobian.get(i, i_hat);
+      numerical_difference_sum.get(i_hat) +=
+          numeric_jacobian_transpose.get(i_hat, i);
+    }
+  }
+  for (size_t i_hat = 0; i_hat < Dim; ++i_hat) {
+    // If the numeric and analytic components are both sufficiently small,
+    // treat them as equal.
+    constexpr double eps_for_ratio = 1.e-10;
+    if (max(abs(jacobian_difference_sum.get(i_hat))) < eps_for_ratio and
+        max(abs(numerical_difference_sum.get(i_hat))) < eps_for_ratio) {
```

Comment:
This check is no longer necessary, because instead I changed this so instead of summing the components, it does sqrt(sum(square))) of the components, so that you don't get zero unless all components of the jacobian are zero. So I no longer do this check at all.